scene/gui/tree.cpp

```diff
@@ -3188,6 +3195,32 @@ bool Tree::_scroll(bool p_horizontal, float p_pages) {
 	return scroll->get_value() != prev_value;
 }
 
+Rect2 Tree::_get_scrollbar_layout_rect() const {
+	const Size2 size = get_size();
+	const Ref<StyleBox> background = theme_cache.panel_style;
+
+	const Rect2 content_rect = Rect2(background->get_offset(), size - background->get_minimum_size());
+
+	const real_t top = theme_cache.scrollbar_margin_top < 0 ? content_rect.get_position().y : theme_cache.scrollbar_margin_top;
+	const real_t right = theme_cache.scrollbar_margin_right < 0 ? content_rect.get_end().x : (size.x - theme_cache.scrollbar_margin_right);
+	const real_t bottom = theme_cache.scrollbar_margin_bottom < 0 ? content_rect.get_end().y : (size.y - theme_cache.scrollbar_margin_bottom);
+	const real_t left = theme_cache.scrollbar_margin_left < 0 ? content_rect.get_position().x : theme_cache.scrollbar_margin_left;
+
+	return Rect2(left, top, right - left, bottom - top);
+}
+
+Size2 Tree::_get_content_rect_offset() const {
+	const Size2 size = get_size();
+	const Ref<StyleBox> background = theme_cache.panel_style;
+	const Point2 content_end = background->get_offset() + size - background->get_minimum_size();
```

Comment:
Same here.

Reply:
In fact, here we can simply take the size and subtract the right-bottom offset, as the top-left is cancelled out.